src/FINDOLOGIC/Api/ResponseObjects/Response.php

```diff
@@ -13,6 +26,37 @@ abstract class Response
      */
     abstract public function __construct($response, $responseTime = null);
 
+    /**
+     * Creates a new Response instance based on the given request builder.
+     *
+     * @param RequestBuilder $requestBuilder
+     * @param GuzzleResponse $response
+     * @param GuzzleResponse|null $alivetestResponse The alivetest response, or null if no alivetest was made.
+     * @param float|null $responseTime
+     * @return Response
+     */
+    public static function getInstance(
+        RequestBuilder $requestBuilder,
+        GuzzleResponse $response,
+        $alivetestResponse = null,
+        $responseTime = null
+    ) {
+        if ($alivetestResponse !== null) {
+            self::checkAlivetestBody($alivetestResponse);
+        }
+        self::checkResponseIsValid($response);
+
+        switch (get_class($requestBuilder)) {
+            case SearchRequestBuilder::class:
+            case NavigationRequestBuilder::class:
+                return new Xml20Response($response->getBody()->getContents(), $responseTime);
+            case SuggestRequestBuilder::class:
+                return new SuggestResponse($response->getBody()->getContents(), $responseTime);
+            default:
+                throw new InvalidArgumentException('Unknown request builder');
```

Comment:
I like to give an indication of the unexpected value in the message since it may make debugging easier, something like `Unknown request build: get_class($requestBuilder)`